### ISSUE TITLE
cannot favourite photos from shared album

### ISSUE THREAD

**Issue by uhthomas** [immich-app/immich#2977]
### The bug

To reproduce:
1. Create a shared album with photos.
2. Try to favourite a photo from the shared album (as a sharee).

![image](https://github.com/immich-app/immich/assets/9749173/a3558b1f-27a7-4ace-a5d6-07c5fe5a8ddd)

### The OS that Immich Server is running on

N/A

### Version of Immich Server

v1.64.0

### Version of Immich Mobile App

1.63.0 build.103

### Platform with the issue

- [X] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
N/A
```

**Comment by uhthomas** [comment]
We spoke a bit and I think the best solution here is to create a new "favourites" table which contains an asset id and a user id. This should allow for favourites to be recorded as desired.

**Comment by jrasm91** [comment]
This bug has been fixed and we can track favorite non owned assets separately. I believe there are discussions about this in GitHub already

**Comment by refactoriel** [comment]
I don't think this is fixed. In v1.134.0 I can favourite a shared photo by clicking the "heart" button. The "heart" state persists, but if I read the response via the dev tools the photo shows up as not favourited. It also doesn't show up in my "favourites" page.

**Comment by Eric-N-Be** [comment]
Also interested in this!!

Shared albums are often used in families or groups of friends. Being able to keep track of favorite partner photos or tag them (e.g. "vacation 2025", "best of") would improve personal organization

Allow users to add favorites and personal tags on photos shared by partners. These favorites/tags should be only visible to the user who added them, not altering the owner’s data.

actually, i can only add favorites foto only if the foto is on my account, it's not possible to had a foto from partners (by web or mobile)

**Comment by krusse** [comment]
Not sure why this is closed. We should be able to add a partner's photo to favourites.

**Comment by alextran1502** [comment]
Part of sharing rework #12614
